### ISSUE TITLE
Feature Request: Touchscreen support

### ISSUE THREAD

**Issue by dianlujitao** [microsoft/terminal#5159]
# Description of the new feature/enhancement

I'm a Surface pro user and sometimes need to use CLI tools without bothering to pick up the keyboard (although it's a mess to type with on-screen keyboard), but it turns out that Windows Terminal is not optimized for touchscreen at all.

# Proposed technical implementation details (optional)

As far as I can tell so far, the following needs to be implemented:

1. Automatically popup on-screen keyboard and resize the window to fit it on touch.
2. Long press to select/copy/paste.
3. Zoom with finger gestures (https://github.com/microsoft/terminal/issues/3149)

**Comment by diligamer** [comment]
Not Useful For Developers

**Comment by DHowett-MSFT** [comment]
@diligamer do you speak for all developers?

**Comment by diligamer** [comment]
@DHowett-MSFT Well, Most Developers Don't Use Touchscreens.

**Comment by zadjii-msft** [comment]
@diligamer Ah yes, but _some_ developers _do_ use touchscreens, so we shouldn't discount the opinion of a minority of users based solely on the fact that they're a minority.

<hr>

Back on topic, I could have sworn that @leonMSFT actually already implemented having the on-screen keyboard appear when you tap on the Terminal. @dianlujitao what verison of the Terminal are you using?

I'm fine with leaving this open to track "Long press to select/copy/paste.", since 
* "Automatically popup on-screen keyboard and resize the window to fit it on touch." _should_ already work
* "Zoom with finger gestures" is already being tracked by #3149.

**Comment by leonMSFT** [comment]
The on-screen keyboard still seems to work as of ver. 0.11.842 on my surface book when i detach my screen FYI!

**Comment by j4james** [comment]
> The on-screen keyboard still seems to work as of ver. 0.11.842 on my surface book when i detach my screen FYI!

Yes! That works for me too. I would have thought just turning on tablet mode might be enough, but this is OK.

However, even with the screen detached, I can't get touch to work for mouse click operations. For example, when using the Linux `dialog` utility, it doesn't seem to register taps on the dialog buttons.

**Comment by dianlujitao** [comment]
Yes, can confirm that on-screen keyboard works after turning on tablet mode as @j4james mentioned

**Comment by FranckPachot** [comment]
touch screen should just act like mouse. Here is, for an example, tmux default mouse actions:
```
bind-key    -T copy-mode    MouseDown1Pane        select-pane
bind-key    -T copy-mode    MouseDrag1Pane        select-pane \; send-keys -X begin-selection                                               bind-key    -T copy-mode    MouseDragEnd1Pane     send-keys -X copy-selection-and-cancel
bind-key    -T copy-mode    WheelUpPane           select-pane \; send-keys -X -N 5 scroll-up
bind-key    -T copy-mode    WheelDownPane         select-pane \; send-keys -X -N 5 scroll-down
bind-key    -T copy-mode    DoubleClick1Pane      select-pane \; send-keys -X select-word
bind-key    -T copy-mode    TripleClick1Pane      select-pane \; send-keys -X select-line
```
I suppose all that should by supported from touchscreen

**Comment by actemium-cincinnati** [comment]
This really kills using Textual or any other mouse oriented console application.

**Comment by blauvster** [comment]
This really kills using Textual or any other touch oriented console application.
Textual looks great in WT and works as expected with a mouse, but since it doesn't support touch I will be switching back to conhost.

**Comment by alisonatwork** [comment]
I just got bitten by this because my mouse is currently not working. The worst part is not just that touching or tapping does absolutely nothing, but when you use the virtual touchpad which gives an on-screen touchpad that allows for navigating using the mouse cursor and left/right click like a regular mouse... still doesn't respond. In fact it responds even less - you can't even click to open the settings menu! At least I would expect the virtual touchpad to work so that we can do everything a mouse can do.

If anyone else is temporarily stuck without a mouse and can't select any more, the best solution I found is to use a bluetooth mouse app on your phone and use the phone screen as a virtual touchpad.

Edit to add: seems like the virtual touchpad does, in fact, work - just not in Administrator mode. So one solution for touchscreen users is to enable the virtual touchpad for selection.

**Comment by ant31** [comment]
> touch screen should just act like mouse. Here is, for an example, tmux default mouse actions:
> 
> ```
> bind-key    -T copy-mode    MouseDown1Pane        select-pane
> bind-key    -T copy-mode    MouseDrag1Pane        select-pane \; send-keys -X begin-selection                                               bind-key    -T copy-mode    MouseDragEnd1Pane     send-keys -X copy-selection-and-cancel
> bind-key    -T copy-mode    WheelUpPane           select-pane \; send-keys -X -N 5 scroll-up
> bind-key    -T copy-mode    WheelDownPane         select-pane \; send-keys -X -N 5 scroll-down
> bind-key    -T copy-mode    DoubleClick1Pane      select-pane \; send-keys -X select-word
> bind-key    -T copy-mode    TripleClick1Pane      select-pane \; send-keys -X select-line
> ```
> 
> I suppose all that should by supported from touchscreen

No the touch screen is not interacting as a mouse on Tmux/terminal for some reason. It doesn't register any action
